wandb/sdk/launch/github_reference.py

```diff
@@ -231,3 +238,40 @@ def _update_path(self, dst_dir: str) -> None:
         elif path.is_dir():
             self.directory = self.path
             self.path = None
+
+    def _pull_repo(self, dst_dir: Optional[str] = None) -> None:
+        """Pull the repo to the destination directory, or to a temp directory if not given.
+
+        :returns: the local directory the repo was pulled to
+        """
+        if dst_dir is None and self.local_dir is not None:
+            return
+        import git
+
+        if dst_dir is None:
+            dst_dir = tempfile.TemporaryDirectory().name
+        self.repo_object = git.Repo.clone_from(self.repo_ssh(), dst_dir)
+        self.local_dir = dst_dir
+
+    def commit(self) -> str:
+        """Get git SHA associated with the reference."""
+        self._pull_repo()
+        assert self.repo_object is not None
+        return self.repo_object.head.commit.hexsha  # type: ignore
+
+    def get_requirements(self) -> str:
+        """Pull requirements.txt from the top-level of the repo."""
+        self._pull_repo()
+        with open(f"{self.local_dir}/requirements.txt") as req_file:
```

Comment:
is this guaranteed? what if the repo doesn't have requirements

Reply:
For now we'll require that a `requirements.txt` is present top-level in the repo. We'll handle `requirements.txt` in subdirectories in the future (likely next week). But good callout, I'll add a more helpful error here.